test/pool-bin/BinPoolManagerBehaviorComparison.t.sol

```diff
@@ -0,0 +1,183 @@
+import {PackedUint128Math} from "../../src/pool-bin/libraries/math/PackedUint128Math.sol";
+
+abstract contract LBFuzzer is LBHelper, BinTestHelper {
+    using BinPoolParametersHelper for bytes32;
+    using PackedUint128Math for bytes32;
+    using PoolIdLibrary for PoolKey;
+
+    IVault vault;
+    IBinPoolManager manager;
+
+    BinLiquidityHelper liquidityHelper;
+    BinSwapHelper swapHelper;
+
+    MockERC20 token0;
+    MockERC20 token1;
+    Currency currency0;
+    Currency currency1;
+
+    function setUp() public virtual override {
+        super.setUp();
+
+        vault = new Vault();
+        manager = new BinPoolManager(vault, 500000);
+        vault.registerApp(address(manager));
+        swapHelper = new BinSwapHelper(manager, vault);
+        liquidityHelper = new BinLiquidityHelper(manager, vault);
+
+        token0 = new MockERC20("TestA", "A", 18);
+        token1 = new MockERC20("TestB", "B", 18);
+        (token0, token1) = token0 < token1 ? (token0, token1) : (token1, token0);
+        currency0 = Currency.wrap(address(token0));
+        currency1 = Currency.wrap(address(token1));
+        token0.mint(address(this), 10 ** 36);
+        token1.mint(address(this), 10 ** 36);
+
+        // add to whitelist required by lb
+        lbFactory.addQuoteAsset(address(token0));
+        lbFactory.addQuoteAsset(address(token1));
+
+        token0.approve(address(swapHelper), type(uint256).max);
+        token1.approve(address(swapHelper), type(uint256).max);
+        token0.approve(address(liquidityHelper), type(uint256).max);
+        token1.approve(address(liquidityHelper), type(uint256).max);
+    }
+
+    function initPools(uint16 binStep, uint24 activeId)
+        public
+        returns (ILBPair lbPair, PoolKey memory key_, uint24 boundActiveId)
+    {
+        binStep = uint16(bound(binStep, manager.MIN_BIN_STEP(), manager.MAX_BIN_STEP()));
+
+        // TODO: bound activeId
+        activeId = uint24(bound(activeId, (1 << 23) - 100, (1 << 23) + 100));
+
+        // lb init
+        lbPair = ILBPair(lbFactory.createLBPair(address(token0), address(token1), activeId, binStep));
+
+        // v4#bin init
+        key_ = PoolKey({
+            currency0: currency0,
+            currency1: currency1,
+            hooks: IHooks(address(0)),
+            poolManager: manager,
+            // we've applied the cl-pool styled fee mechanism to bin-pool
+            // which is completely different from existing one
+            // hence set fee to 0 to avoid interference
+            fee: 0,
+            parameters: bytes32(0).setBinStep(binStep)
+        });
+        manager.initialize(key_, activeId, "");
+
+        boundActiveId = activeId;
+    }
+
+    function mint(ILBPair lbPair, PoolKey memory key, uint24 activeId, uint256 amountX, uint256 amountY, uint8 binNum)
+        public
+    {
+        // TODO: investigate why the following line fails the test
+        // binNum = uint8(bound(binNum, 1, 20));
+        binNum = 5;
```

Comment:
Not urgent, but if u have time pls help to take a look @ChefMist . I still didn't figure out why uncomment this will line will fail the test

Reply:
1. Transfer amountX / amountY to lbpair not balanceDelta.amount0 eg.

```
token0.transfer(address(lbPair), uint256(int256(amountX)));
token1.transfer(address(lbPair), uint256(int256(amountY)));
```

2. to assert  total amt0/amt1 used in minted maybe can do something like

```
// eg. to compare amount0 
from v4: balanceDelta.amount0 
from v3: amountX - amountLeft.decodeX() 
```

can you try this and see if it works?

Reply:
the above works as bin pool see whats the `amountX` and `amountY` and then divided by the liquidityParam to know how much amount to bin. 

The current test setup could lead to  potentially `1000` for v4 but `995` for v3 as input for amount0

Reply:
oh, it worked ! thx for helping